tokens/usecase/pricing/pricing.go

```diff
@@ -7,13 +7,17 @@ import (
 	"github.com/osmosis-labs/sqs/domain/cache"
 	"github.com/osmosis-labs/sqs/domain/mvc"
 	chainpricing "github.com/osmosis-labs/sqs/tokens/usecase/pricing/chain"
+	coingeckopricing "github.com/osmosis-labs/sqs/tokens/usecase/pricing/coingecko"
 )
 
 // NewPricingStrategy is a factory method to create the pricing strategy based on the desired source.
 func NewPricingStrategy(config domain.PricingConfig, tokensUsecase mvc.TokensUsecase, routerUseCase mvc.RouterUsecase) (domain.PricingSource, error) {
 	if config.DefaultSource == domain.ChainPricingSourceType {
 		return chainpricing.New(routerUseCase, tokensUsecase, config), nil
 	}
+	if config.DefaultSource == domain.CoinGeckoPricingSourceType {
+		return coingeckopricing.New(routerUseCase, tokensUsecase, config, coingeckopricing.DefaultCoingeckoPriceGetter), nil
+	}
```

Comment:
It feels odd that we pass in an entire config while only needing the `DefaultSource` field.

Should we instead pass in a `domain.CoinGeckoPricingSourceType` variable?

Reply:
But we still need the `config` when calling the `New` in `chainpricing.New` and `coingeckopricing.New`

Reply:
You're right. I must have missed something - apologies